internal/block/guarantee.go

```diff
@@ -157,14 +157,22 @@ func NewErrorWorkResult(serviceId ServiceId, serviceHashCode, payloadHash crypto
 }
 
 func (w *WorkReport) Hash() (crypto.Hash, error) {
+	jsonData, err := w.Encode()
```

Comment:
This variable name is misleading, can you rename it to something else? (eg `encodedData`)